### PR TITLE
sweethome3d: fix startup issue and upgrade JDK

### LINKED ISSUE #270010: Sweet Home 3D not starting
### Describe the bug
Wanted to install and start sweethomed.application. the build worked without problems but when i started it, it didnt started.

### Steps To Reproduce
1. install sweethome3d.application via configuration file
2. try to start it

### Expected behavior
Starting the Application

### Screenshots
stucks at the start screen
Screenshot_20231126_003558

### Additional context
At the moment its installed in my user context. When i try to start it as my user over the terminal i get no errors. When i start it with "sudo sweethome3d" I get the following error, and it also dont start, also start at starting screen and vanishing from the task bar:

```
[xyz00777@NCC-1031:~]$ sudo sweethome3d
[sudo] password for xyz00777: 
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.eteks.sweethome3d.SweetHome3DBootstrap.main(Unknown Source)
Caused by: java.lang.ExceptionInInitializerError
        at javax.media.j3d.GraphicsConfigTemplate3D.isGraphicsConfigSupported(GraphicsConfigTemplate3D.java:348)
        at com.eteks.sweethome3d.j3d.Component3DManager.createGraphicsConfigurationTemplate3D(Unknown Source)
        at com.eteks.sweethome3d.j3d.Component3DManager.<init>(Unknown Source)
        at com.eteks.sweethome3d.j3d.Component3DManager.getInstance(Unknown Source)
        at com.eteks.sweethome3d.SweetHome3D.addComponent3DRenderingErrorObserver(Unknown Source)
        at com.eteks.sweethome3d.SweetHome3D.init(Unknown Source)
        at com.eteks.sweethome3d.SweetHome3D.main(Unknown Source)
        ... 5 more
Caused by: com.jogamp.opengl.GLException: Profiles [GL4bc, GL3bc, GL2, GLES1] not available on device null
        at com.jogamp.opengl.GLProfile.get(GLProfile.java:1057)
        at com.jogamp.opengl.GLProfile.get(GLProfile.java:1068)
        at com.jogamp.opengl.GLProfile.getMaxFixedFunc(GLProfile.java:821)
        at javax.media.j3d.JoglPipeline.initialize(JoglPipeline.java:131)
        at javax.media.j3d.Pipeline.createPipeline(Pipeline.java:92)
        at javax.media.j3d.MasterControl.loadLibraries(MasterControl.java:858)
        at javax.media.j3d.VirtualUniverse.<clinit>(VirtualUniverse.java:267)
        ... 12 more

```
I think these happens because its not installed as user, but propably it helps to find the issue.

### Notify maintainers
@edwtjo

### Metadata

```console
[xyz00777@NCC-1031:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.63, NixOS, 23.11 (Tapir), 23.11pre551191.51a01a7e5515`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(xyz00777): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by nbdd0121**
## Description of changes

Fix SH3D startup issue caused by inability to find libGL.

Fixes #270010

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by nbdd0121** [comment]
Result of `nixpkgs-review pr 284982` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>sweethome3d.application</li>
    <li>sweethome3d.furniture-editor</li>
    <li>sweethome3d.textures-editor</li>
  </ul>
</details>

**Review by WilliButz — APPROVED** [review]
LGTM, tested on sway :+1: 

Just stumbled over `mainProgram`: in my opinion it seems completely fine here to reference an attribute also used to reference the executable in the buildPhase. I saw that #281490 wants to enforce the setting to be a hard-coded string. Maybe the documented requirements for `mainProgram` don't need to be so strict.

File: pkgs/applications/misc/sweethome3d/default.nix [hunk 1]
```diff
@@ -3,16 +3,16 @@
 , fetchurl
 , makeWrapper
 , makeDesktopItem
-# sweethome3d 6.5.2 does not yet fully build&run with jdk 9 and later?
-, jdk8
-, jre8
+, jdk
+, jre
```
Comment (Tungsten842) [inline pkgs/applications/misc/sweethome3d/default.nix:7]: Jre and jdk are the same thing:
https://github.com/NixOS/nixpkgs/blob/1a4d722e81aa9156a3028f6b3846d2156b3a4f76/pkgs/top-level/all-packages.nix#L16441-L16448

Comment (nbdd0121) [reply pkgs/applications/misc/sweethome3d/default.nix]: Fixed

**Pull request merged**